### PR TITLE
remove 0.6 deprecations

### PR DESCRIPTION
**Pull request by Sacha0**
This pull request removes (almost all) 0.6 deprecations. (Three or so deprecations remain; I hope to remove those tomorrow but might need a pointer or two.) Let's see what CI says. Best!

### CODE REVIEW

File: base/client.jl [hunk 1]
```diff
@@ -223,10 +223,6 @@ function parse_input_line(s::String; filename::String="none")
     # expr
     ex = ccall(:jl_parse_input_line, Any, (Ptr{UInt8}, Csize_t, Ptr{UInt8}, Csize_t),
                s, sizeof(s), filename, sizeof(filename))
-    if ex isa Symbol && all(equalto('_'), string(ex))
-        # remove with 0.7 deprecation
-        Meta.lower(Main, ex)  # to get possible warning about using _ as an rvalue
-    end
```
Comment (JeffBezanson) [inline base/client.jl]: I think this one needs to stay.

Comment (Sacha0) [reply base/client.jl]: Good catch! Fixed. Thanks! :)

**Comment by KristofferC** [comment]
Dates stdlib has a deprecation file.

**Comment by Sacha0** [comment]
> Dates stdlib has a deprecation file.

Good catch! The IterativeEigensolvers and SuiteSparse deprecations are for 0.7 (I should add a note), and going by the note at the top of the file I *think* the Dates deprecations are also for 0.7. Please correct me if I am incorrect!

**Comment by KristofferC** [comment]
Right, I forgot I removed the 0.6 deprecations for Dates.

**Comment by JeffBezanson** [comment]
There are also a few in the front end; I'll try to get to those.

**Comment by JeffBezanson** [comment]
Ah yes, of course --- this will require fixing Documenter.

**Comment by Sacha0** [comment]
Indeed so :). I plan to restore the problematic deprecations here and address them separately. (Apologies for the absence of motion on this front! Focusing on #25364.)

**Comment by Sacha0** [comment]
Rebased and fixed the doc build failure. What say you now, CI?

**Comment by Sacha0** [comment]
The AV failures appear unrelated (one timeout after successfully completing the main test suite, one remote exception deserialization failure in the `pkg` tests after successfully completing the main test suite). The CircleCI failure appears unrelated as well (a configuration issue?). So CI generally seems happy now.

Perhaps someone could give this pull request a final sweep and merge? Absent that and objections, I plan to merge this pull request Sunday evening PT at the earliest and pick off the tricker remaining removals in downstream work (unless someone happily beats me to it! :) ). Best!

**Comment by Sacha0** [comment]
Rebased. Absent objections or requests for time, I plan to merge these changes once CI clears again. Best!

File: base/deprecated.jl [hunk 2]
```diff
@@ -827,291 +184,6 @@ _promote_eltype_op(op, A, B, C, D...) = (@_inline_meta; _promote_eltype_op(op, e
...
-function Matrix{T}() where T
-    zero_arg_matrix_constructor("Matrix{T}")
-    return Matrix{T}(uninitialized, 0, 0)
-end
-function Matrix()
-    zero_arg_matrix_constructor("Matrix")
-    return Matrix(uninitialized, 0, 0)
-end
-
-# TODO: remove warning for using `_` in parse_input_line in base/client.jl
```
Comment (tkelman) [inline base/deprecated.jl]: this todo wasn't done, should be put back until it is

Comment (Sacha0) [reply base/deprecated.jl]: Ref. https://github.com/JuliaLang/julia/pull/25312#discussion_r159025396 :).

Comment (tkelman) [reply base/deprecated.jl]: one underscore was deprecated in 0.6. all-underscores in 0.7.

Comment (Sacha0) [reply base/deprecated.jl]: You are suggesting to modify the linked check such that only multiple-underscore instances trigger a warning?

Comment (JeffBezanson) [reply base/deprecated.jl]: Not urgent though, since we haven't yet reclaimed `_` for anything else.

Comment (tkelman) [reply base/deprecated.jl]: Delaying the warning-into-error removal of the single underscore case from 0.7 to 1.0 is probably fine. But the TODO about client.jl should at least be put back in the 0.7 section so it doesn't get forgotten and left as only a warning in 1.0.

Comment (Sacha0) [reply base/deprecated.jl]: I assumed existence of a relevant TODO in the 0.7 section. If no such TODO exists, then indeed, moving this TODO to the 0.7 section sounds wise. I'll check and whip together a pull request if necessary. Thanks! :)

**Pull request merged**